Rules/UseProcessBlockForPipelineCommand.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class UseProcessBlockForPipelineCommand : IScriptRule
+    {
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            IEnumerable<Ast> functionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);
+
+            foreach (FunctionDefinitionAst funcAst in functionAsts)
+            {
+                if
+                (
+                    funcAst.Body.ParamBlock == null
+                    || funcAst.Body.ParamBlock.Attributes == null
+                    || funcAst.Body.ParamBlock.Parameters == null
+                    || funcAst.Body.ProcessBlock != null
+                )
+                { continue; }
+                
+                foreach (var paramAst in funcAst.Body.ParamBlock.Parameters)
+                {
+                    foreach (var paramAstAttribute in paramAst.Attributes)
+                    {
+                        if (!(paramAstAttribute is AttributeAst)) { continue; }

```

Comment:
```suggestion
                        if (!(paramAstAttribute is AttributeAst paramAttributeAst)) { continue; }
```

Reply:
I think this conflicts with the previous comment

resolved in c7a2531c